javalib/src/main/scala/java/util/NullRejectingHashMap.scala

```diff
@@ -57,6 +57,12 @@ private[util] class NullRejectingHashMap[K, V](
     super.put(key, value)
   }
 
+  override def putIfAbsent(key: K, value: V): V = {
+    if (key == null)
+      throw new NullPointerException()
```

Comment:
Added assertions https://github.com/scala-js/scala-js/pull/4367/commits/fa1678d61c4b6f19dfee95000e95c28b9ff135b0
It seems that `putIfAbsent` already throws when `null` value.

Reply:
Well, in `ConcurrentHashMap` it does, because you it still has the old implementation rather than forwarding to this one.

Reply:
In fact, now that I look at it, the implementation should be moved for `remove` and `replace` as well.

Reply:
Added assertions in https://github.com/scala-js/scala-js/pull/4367/commits/d1aa9be1f7713c983a60c2baebaba10444c4d27d
It seems that `remove` and `replace` already throws  NPE, so overriding is seems not required.

Reply:
They only throw an NPE because that's what the implementation in ConcurrentHashMap does. You need to change the implementations in ConcurrentHashMap to simply forward to NullRejectingHashMap. (that applies to `putIfAbsent`, `remove` and `replace`). Then you'll see that `remove` and `replace` also need updating.

Reply:
Moved implementation to NullRejectingHashMap in https://github.com/scala-js/scala-js/pull/4367/commits/811bd513c9b9a461ddb88969460173663d981de3